pow/web3j/src/main/java/org/ethereum/beacon/pow/Web3JTransactionBuilder.java

```diff
@@ -0,0 +1,117 @@
+  private final Web3j web3j;
+  private final Address contractDeployAddress;
+  private final Web3RequestExecutor web3RequestExecutor;
+
+  public Web3JTransactionBuilder(Web3j web3j, String contractDeployAddress) {
+    this.web3j = web3j;
+    this.contractDeployAddress = Address.fromHexString(contractDeployAddress);
+    this.web3RequestExecutor = new Web3RequestExecutor(web3j);
+  }
+
+  @Override
+  public CompletableFuture<BytesValue> createTransaction(
+      String fromAddress, DepositData depositData, Gwei amount) {
+    CompletableFuture<BytesValue> result = new CompletableFuture<>();
+    web3RequestExecutor.executeOnSyncDone(
+        () -> {
+          CompletableFuture<EthGetTransactionCount> txCountFut =
+              web3j
+                  .ethGetTransactionCount(fromAddress, DefaultBlockParameterName.LATEST)
+                  .sendAsync();
+          CompletableFuture<EthGasPrice> gasPriceFut = web3j.ethGasPrice().sendAsync();
+          CompletableFuture.allOf(txCountFut, gasPriceFut)
+              .thenAccept(
+                  aVoid -> {
+                    BigInteger nonce = txCountFut.join().getTransactionCount();
+                    BigInteger value =
+                        Convert.toWei(amount.longValue() + "", Convert.Unit.GWEI).toBigInteger();
+                    BigInteger gasLimit = BigInteger.valueOf(2_000_000);
+                    final Function function =
+                        new Function(
+                            FUNC_DEPOSIT,
+                            Arrays.asList(
+                                new org.web3j.abi.datatypes.DynamicBytes(
+                                    depositData.getPubKey().extractArray()),
+                                new org.web3j.abi.datatypes.DynamicBytes(
+                                    depositData.getWithdrawalCredentials().extractArray()),
+                                new org.web3j.abi.datatypes.DynamicBytes(
+                                    depositData.getSignature().extractArray())),
+                            Collections.emptyList());
+                    String encodedFunction = FunctionEncoder.encode(function);
+                    byte[] rawTransaction =
+                        RlpEncoder.encode(
+                            new RlpList(
+                                RlpString.create(nonce),
+                                RlpString.create(gasPriceFut.join().getGasPrice()),
+                                RlpString.create(gasLimit),
+                                RlpString.create(contractDeployAddress.toString()),
+                                RlpString.create(value),
+                                RlpString.create(encodedFunction)));
+                    result.complete(BytesValue.wrap(rawTransaction));
+                  });
+        });
+
+    return result;
+  }
+
+  @Override
+  public CompletableFuture<BytesValue> signTransaction(
+      BytesValue unsignedTransaction, BytesValue eth1PrivKey) {
```

Comment:
We might want to avoid private key management and dedicate it to geth or parity. Any thoughts on that?

Reply:
I've created issue #169